controllers/tykoasapidefinition_controller.go

```diff
@@ -188,6 +224,192 @@ func (r *TykOasApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.
 	return ctrl.Result{RequeueAfter: reqA}, err
 }
 
+// reconcileVersionedOasApi checks if a versioned API can be deleted and isn't a dangling Versioned API
+// i.e an API that was previously part of the version list of a Base API.
+func (r *TykOasApiDefinitionReconciler) reconcileVersionedOasApi(
+	ctx context.Context,
+	log logr.Logger,
+	versionedTykOAS *v1alpha1.TykOasApiDefinition,
+) error {
+	if err := r.cleanUpDanglingVersionedOasAPIs(ctx, versionedTykOAS, log); err != nil {
+		return err
+	}
+
+	if !versionedTykOAS.ObjectMeta.DeletionTimestamp.IsZero() {
+		if versionedTykOAS.Status.GetIsDefaultVersion() {
+			errMsg := "Cannot delete default API version. Remove it from base API's default first"
+
+			versionedTykOAS.Status.LatestTransaction = v1alpha1.TransactionInfo{
+				Time:   metav1.Now(),
+				Status: v1alpha1.TransactionStatus(errMsg),
+			}
+
+			log.Info("error deleting versioned TykOasAPI", errMsg)
```

Comment:
What do you think updating this log to Error? Are there any specific reason to print Info log here instead of Error?

Reply:
The reason for `log.Info` is that the error isn't a critical one, it more of a validation type of error. The error log often looks like something critical happened.

Reply:
But user is trying to delete a CR and it is not getting deleted. Shouldn't it logged as error as operation is unsuccessful?